src/Microsoft.Azure.SignalR.Protocols/ServiceMessage.cs

```diff
@@ -12,6 +14,49 @@ public abstract class ServiceMessage
     {
     }
 
+    /// <summary>
+    /// Base class of messages between Azure SignalR Service and SDK.
+    /// </summary>
+    public abstract class ExtensibleServiceMessage : ServiceMessage
```

Comment:
what if I only need extensible to `ConnectionDataMessage`? just have `ExtensibleServiceMessage` to `ServiceMessage` as TraceId sounds applicable for all the messages?

Reply:
yes, leave it to @wanlwanl 😃